src/iuse.cpp

```diff
@@ -2661,39 +2656,28 @@ int iuse::sew_advanced(player *p, item *it, bool, const tripoint& )
         return 0;
     }
 
-    itype_id repair_item = "none";
     std::vector<std::string> plurals;
     std::vector<itype_id> repair_items;
     std::string plural = "";
     //translation note: add <plural> tag to keep them unique
-    if (mod->made_of("cotton")) {
-        repair_items.push_back("rag");
-        plurals.push_back(rm_prefix(_("<plural>rags")));
-    }
-    if (mod->made_of("leather")) {
-        repair_items.push_back("leather");
-        plurals.push_back(rm_prefix(_("<plural>leather")));
-    }
-    if (mod->made_of("fur")) {
-        repair_items.push_back("fur");
-        plurals.push_back(rm_prefix(_("<plural>fur")));
-    }
-    if (mod->made_of("nomex")) {
-        repair_items.push_back("nomex");
-        plurals.push_back(rm_prefix(_("<plural>Nomex")));
-    }
-    if (mod->made_of("plastic")) {
-        repair_items.push_back("plastic_chunk");
-        plurals.push_back(rm_prefix(_("<plural>plastic")));
-    }
-    if (mod->made_of("kevlar")) {
-        repair_items.push_back("kevlar_plate");
-        plurals.push_back(rm_prefix(_("<plural>Kevlar")));
-    }
-    if (mod->made_of("wool")) {
-        repair_items.push_back("felt_patch");
-        plurals.push_back(rm_prefix(_("<plural>wool")));
-    }
+
+    // Little helper to cut down some surplus redundancy and repetition
+    const auto add_material = [&]( const itype_id &material,
+                                   const itype_id &mat_item,
+                                   const std::string &plural ) {
+        if( mod->made_of( material ) ) {
+            repair_items.push_back( mat_item );
+            plurals.push_back( rm_prefix( _( plural.c_str() ) ) );
+        }
+    };
+
+    add_material( "cotton", "rag", "<plural>rags" );
```

Comment:
The `_()` needs to be around the literal string, otherwise it will not be extracted for translation.